rust/src/cliwrap/yumdnf.rs

```diff
@@ -0,0 +1,152 @@
+// SPDX-License-Identifier: Apache-2.0 OR MIT
+
+use anyhow::{anyhow, Result};
+use indoc::indoc;
+use std::os::unix::process::CommandExt;
+use std::process::Command;
+use structopt::StructOpt;
+
+const UNHANDLED: &str = indoc::indoc! {r#"
+rpm-ostree: Unknown arguments passed to yum/dnf compatibility wrapper.
+
+Please see `yum --help` for currently accepted commands via this
+wrapper, and `rpm-ostree --help` for more information about.the rpm-ostree
+project.
+"#};
+
+#[derive(Debug, StructOpt)]
+#[structopt(
+    name = "yumdnf-rpmostree",
+    about = "Compatibility wrapper implementing subset of yum/dnf CLI"
+)]
+#[structopt(rename_all = "kebab-case")]
+/// Main options struct
+enum Opt {
+    /// Start an upgrade of the operating system
+    Upgrade,
```

Comment:
Also one thing that's clear to me is we can't really use structopt for this going forward - what we need is something where we can dynamically pass through arguments like `yum upgrade --help` to be `rpm-ostree upgrade --help` and print that help text.  And also relatedly, not replicate all of the arguments here.